tests/Unit/IO/Test_H5.py

```diff
@@ -116,6 +116,46 @@ def test_groups(self):
             self.assertTrue(group_name in file_spec.groups())
         file_spec.close()
 
+    def test_exceptions(self):
+        file_spec = spectre_h5.H5File(file_name=self.file_name,
+                                      append_to_file=True)
+
+        #create existing file
+        with self.assertRaises(RuntimeError):
+            spectre_h5.H5File(file_name=self.file_name, append_to_file=False)
+
+        file_spec.insert_dat(path="/element_data",
+                             legend=["Time", "Value"],
+                             version=0)
+
+        #insert existing data file
+        with self.assertRaises(RuntimeError):
+            file_spec.insert_dat(path="/element_data",
+                                 legend=["Time", "Value"],
+                                 version=0)
+
+        # grab non-existing data file
+        try:
+            with self.assertRaisesRegex(RuntimeError, "element_data.dat"):
+                file_spec.get_vol("/element_dat")
+        except AttributeError:
```

Comment:
Instead of this `try`/`except`, add this code below the imports at the beginning of the file:

```py
# For Py2 compatibility
try:
    unittest.TestCase.assertRaisesRegex
except AttributeError:
    unittest.TestCase.assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
```